woocommerce/woocommerce-gutenberg-products-block
# Product Gallery Thumbnails: Add View all overlay

This PR adds the "View all" overlay to the last thumbnail which triggers the lightbox/dialog (which will display all the product gallery images).



## What

Fixes #11083

## Why

The "View all" overlay provides the user with feedback about the total number of available product images in the gallery (and offer a way to view them). 

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. With a block theme, open `Appearance > Editor`.
2. Open `Templates > Single Product`.
3. Select `Convert to Blocks` if you are using the Classic Templates.
4. Add the `Product Gallery` block.
5. Ensure the Thumbnails are enabled.
6. Open a product page on the frontend **with multiple product images** (more than `Number of Thumbnails` setting) and test the frontend. Make sure the last thumbnail displays an overlay that opens the Dialog on click.
7. Make sure the overlay displays the correct number of remaining thumbnails.
8. Make sure there are no regressions and the Dialog can still be triggered by clicking the Large Image.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|<img width="539" alt="V-Neck_T-Shirt_–_newproductgallery" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/7c2b9764-6bb9-462d-8d21-bcedc469bcd4">|<img width="543" alt="V-Neck_T-Shirt_–_newproductgallery" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/f28faeab-1d81-4b6d-86a5-860c3af4cb85">|

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Product Gallery Thumbnails: Add a View all overlay.
